talent/job-descriptions/intern-ds.md

```diff
@@ -0,0 +1,68 @@
+
+- source{d} tries to be as open in it's ongoing research as possible. If a handy developed tool can be open-sourced then it has to be. If a solved interesting problem can be disclosed, it has to be published in source{d}'s blog. If a funny dataset is compiled, it is published on data.world. Writing papers is welcomed. We talk on conferences literally every month.
+
+- To this moment, source{d}'s engineers have developed:
+- src-d/awesome-machine-learning-on-source-code - everything that we know about MLoSC.
+- ml - MLoSC framework.
+- apollo - modular source code de-duplication research project.
+- kmcuda - lightning fast K-means and K-nearest neighbours on NVIDIA GPUs. It allows us to cluster 3 million samples, each 256 dimensions, into 1000 clusters in less than 20 minutes with the outstanding precision using two NVIDIA Titan X cards.
+- minhashcuda - lightning fast Weighted MinHash on NVIDIA GPUs. This tool allowed us to scan all +17M git repositories in 30 minutes and find 1.5M duplicates (the results were
+published on data.world.
+- lapjv - Jonker-Volgenant algorithm to solve linear sum assignment problems, accelerated with Intel AVX2 instruction set. It is capable of transforming 4096 t-SNE dots into 64x64 images in 2 minutes.
+- wmd-relax - optimized Word Mover's Distance
+- hercules - Git repositories line burn down analysis command line tool built on top of source{d}'s go-git - Git client and server implementation in pure Go language.
+• sparkpickle - the tool to read PySpar RDD files without having to install Spark.
+
+### The following notable technical posts, papers and talks exist:
+- Source Code Identifier Embeddings
+- Open Source Stack for Machine Learning on Source Code
+- Analyzing GitHub, How Developers Change Programming Languages Over Time
+- GitHub Contributions Graph: Analyzing Pagerank & Proving the 6 Handshakes Theory
+- Similarity of GitHub Repositories by Source Code Identifiers
+- 397 Languages, 18,000,000 GitHub repositories, 1.2 billion files, 20 terabytes of code: Spaces or Tabs
+- Topic modeling of public repositories at scale using names in source code - paper
+- Hands on with the most starred GitHub repositories
+- Source code abstracts classification using CNN
+
+### The following datasets published:
+- Names in source code extracted from 13 000 000 GitHub repositories. Not people! 30GB
+- October 2016 GitHub repositories not marked as forks but very similar to each other
+- Readme files found in all GitHub repositories (16M, October 2016
+- ≈ 452,000,000 commits' metadata taken from 16,000,000 repositories on GitHub (Oct 2016)
+
+
+ROLE
+
+- Strong computer science and machine learning background is essential for data science team members. You will be expected to be a passionate, skilful engineer who is able to produce amazing results quickly and reliably. Coding skills are important; we are using Python 3 in our research and production prototyping. Besides, we occasionally code in Go, C++ and CUDA.
+
```

Comment:
MLoSC shares common ideas with Natural Language Processing, so solid **NLP** knowledge is also required. **Deep Learning** experience is highly appreciated.